packages/app/src/cli/services/app-logs/ui/components/Logs.tsx

```diff
@@ -0,0 +1,207 @@
+
+import {Static, Box, Text} from '@shopify/cli-kit/node/ink'
+
+export interface LogsProps {
+  logsProcess: ({
+    jwtToken,
+    cursor,
+    filters,
+  }: {
+    jwtToken: string
+    cursor?: string | undefined
+    filters?: {
+      status?: string
+      source?: string
+    }
+  }) => Promise<{
+    cursor?: string | undefined
+    errors?: string[] | undefined
+    appLogs?: AppEventData[] | undefined
+  }>
+  developerPlatformClient: DeveloperPlatformClient
+  storeId: string
+  apiKey: string
+  cursor: string
+  jwtToken: string
+  filters?: {
+    status?: string
+    source?: string
+  }
+}
+
+interface DetailsFunctionRunLogEvent {
+  input: string
+  inputBytes: number
+  invocationId: string
+  output: string
+  outputBytes: number
+  logs: string
+  functionId: string
+  fuelConsumed: string
+  errorMessage: string | null
+  errorType: string | null
+  status: string
+  source: string
+  eventType?: string
+}
+
+const POLLING_INTERVAL_MS = 450
+const POLLING_ERROR_RETRY_INTERVAL_MS = 5 * 1000
+const POLLING_THROTTLE_RETRY_INTERVAL_MS = 60 * 1000
+const ONE_MILLION = 1000000
+
+const Logs: FunctionComponent<LogsProps> = ({
+  logsProcess,
+  filters,
+  cursor,
+  jwtToken,
+  developerPlatformClient,
+  storeId,
+  apiKey,
+}) => {
+  const pollingInterval = useRef<NodeJS.Timeout>()
+  const currentIntervalRef = useRef<number>(POLLING_INTERVAL_MS)
+  const [logs, setLogs] = useState<DetailsFunctionRunLogEvent[]>([])
+  const [errorsState, setErrorsState] = useState<string[]>([])
+  const [jwtTokenState, setJwtTokenState] = useState<string | null>(jwtToken)
+
+  const pollLogs = async (currentCursor: string) => {
+    try {
+      if (jwtTokenState === null) {
+        // Resubscribe here if needed
+        const jwtToken = await subscribeProcess({
+          logsConfig: {
+            developerPlatformClient,
+            storeId,
+            apiKey,
+          },
+        })
+        if (!jwtToken) {
+          return
+        }
+        setJwtTokenState(jwtToken.jwtToken)
+      }
+      const {
+        cursor: newCursor,
+        errors,
+        appLogs,
+      } = await logsProcess({jwtToken: jwtTokenState || jwtToken, cursor: currentCursor, filters})
+      if (errors) {
+        if (errors.some((error) => error.includes('429'))) {
+          currentIntervalRef.current = POLLING_THROTTLE_RETRY_INTERVAL_MS
+          setErrorsState([...errors, `Retrying in ${POLLING_THROTTLE_RETRY_INTERVAL_MS / 1000} seconds.`])
+        } else if (errors.some((error) => error.includes('401'))) {
+          setJwtTokenState(null)
+          setErrorsState([...errors, 'Resubscribing to logs.'])
+        } else {
+          currentIntervalRef.current = POLLING_ERROR_RETRY_INTERVAL_MS
+          setErrorsState([...errors, `Retrying in ${POLLING_ERROR_RETRY_INTERVAL_MS / 1000} seconds.`])
+        }
+      }
+
+      if (newCursor) {
+        setErrorsState([])
+        currentIntervalRef.current = POLLING_INTERVAL_MS
+      }
```

Comment:
Rather than an if here, should we just put an early return if there was an error?

Reply:
Yea, just relizing I dont even think we need this check. If there is no cursor, then that error would have been thrown in the polling process, so would exit the terminal (since its a 400 error we dont handle).